backends/arm/passes/annotate_channels_last_dim_order_pass.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from executorch.backends.arm.tosa_quant_utils import dq_op
+from executorch.backends.arm.tosa_utils import is_consumer_node_depthwise_conv2d
+from executorch.exir.pass_base import ExportPass, PassResult
+
+
+class AnnotateChannelsLastDimOrder(ExportPass):
+    """
+    Annotates each node with a tosa_dim_order. tosa_dim_order can be seen as a channels-last dim-order
+    that in most cases will be (0, 2, 3, 1) for nodes with 4D-shapes.
+    The annotated tosa_dim_order is used to permute the node's shape such that it
+    gives a TOSA-compliant shape.
+    """
+
+    def is_weight_node_for_dw_conv(self, node: torch.fx.Node):
+        """
+        returns True for dq and w in the following sequences;
+        w -> dw_conv -> ...
+        w -> dq -> dw_conv -> ...
+        """
+        if node.op == "call_function":
+            if node.target != dq_op:
+                return False
+            prev_node = node.args[0]
+            if prev_node.op != "placeholder":
+                return False
+            return is_consumer_node_depthwise_conv2d(node)
+        elif node.op == "placeholder":
+            # node is an input, weight or bias node
+            consumer_node = list(node.users)[0]
+            if self.is_weight_node_for_dw_conv(consumer_node):
```

Comment:
nit: use consistent naming for depthwise convolution helper functions?

Reply:
Updated in last rebase.